src/xtopcom/xvledger/src/xvblockbuild.cpp

```diff
@@ -917,13 +917,21 @@ namespace top
 #endif
             std::string parent_cert_bin;
             parent->get_cert()->serialize_to_string(parent_cert_bin);
-            xmerkle_t<utl::xsha2_256_t, uint256_t> merkle(out_leafs);
+            xmerkle_t<utl::xsha2_256_t, uint256_t> merkle;
+            //test block version
+            if (parent->get_block_version() > base::enum_xvblock_fork_version_unit_opt) {
+                merkle.calc_root_hash(out_leafs, false);
+            }else{
+                merkle.calc_root_hash(out_leafs);
+            }
+            int leaf_index = 0;
             for (auto & _unit : units) {
```

Comment:
for (auto & _unit : units)   change to for (leaf_index=0; leaf_index < count; leaf_index++)